apis/projectcontour/v1/httpproxy.go

```diff
@@ -314,6 +314,10 @@ type TimeoutPolicy struct {
 	Idle string `json:"idle,omitempty"`
 }
 
+// RetryOn is a string type alias with validation to ensure that the value is valid.
+// +kubebuilder:validation:Enum="5xx";gateway-error;reset;connect-failure;retriable-4xx;refused-stream;retriable-status-codes;retriable-headers
```

Comment:
The Envoy protobuf docs say that this accepts both `x-envoy-retry-on` and `x-envoy-retry-grpc-on` values. Should we add the `x-envoy-retry-grpc-on` values to the enum here as well?

Reply:
Looking at Envoy's code, it appears that retrying gRPC behaves independently of any other configuration:
https://github.com/envoyproxy/envoy/blob/master/source/common/router/config_impl.cc#L74-L75
https://github.com/envoyproxy/envoy/blob/master/source/common/router/retry_state_impl.cc#L323-L342

So yes, this should be harmless to add.